### ISSUE TITLE
Create "materialized" ontology for AmiGO /GOlr (go-gaf)

### ISSUE THREAD

**Issue by kltm** [geneontology/go-ontology#19120]
It would be nice to look at the current AmiGO load

"http://skyhook.berkeleybop.org/master/ontology/extensions/go-gaf.owl",
"http://skyhook.berkeleybop.org/master/ontology/extensions/gorel.owl",
"http://skyhook.berkeleybop.org/master/ontology/extensions/go-modules-annotations.owl",
"http://skyhook.berkeleybop.org/master/ontology/extensions/go-taxon-subsets.owl",
"http://purl.obolibrary.org/obo/eco/eco-basic.owl",
"http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl",
"http://purl.obolibrary.org/obo/cl/cl-basic.owl",
"http://purl.obolibrary.org/obo/pato.owl",
"http://purl.obolibrary.org/obo/po.owl",
"http://purl.obolibrary.org/obo/chebi.owl",
"http://purl.obolibrary.org/obo/uberon/basic.owl",
"http://purl.obolibrary.org/obo/wbbt.owl"

It would be good to do for AmiGO what we've done for Minerva and the Noctua AC with go-lego.owl: make a single "materialized" ontology that can be easily reused within the pipeline (and for end users).

**Comment by balhoff** [comment]
Could this be an enhancement to go-gaf? Or, what other uses does go-gaf have besides being loaded here?

**Comment by kltm** [comment]
@balhoff Definitely. I'm not sure there are actually other uses--the "gaf" and "lego" parts referring to their intended data context. That said, the overall load has accrued some other ontologies. Ideally, we could trim the fat and be left with a single product, whether something new or just correcting go-gaf.

**Comment by ukemi** [comment]
I would be a fan of one ontology to bind them all. I've never really been a big fan of having different ontologies for different purposes.

**Comment by kltm** [comment]
@balhoff After being pinged by @dougli1sqrd , it looks like go-gaf.owl is already materialized. That means that only the rationalization between go-lego and go-gaf needs to be accomplished.

**Comment by kltm** [comment]
go-gaf itself is:
http://purl.obolibrary.org/obo/go/extensions/gorel.owl
http://purl.obolibrary.org/obo/cl/cl-basic.owl
http://purl.obolibrary.org/obo/go/extensions/go-plus.owl
http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim.owl
http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim-disjoint-over-in-taxon.owl

---
So:

Exclusive to go-gaf:
http://purl.obolibrary.org/obo/go/extensions/go-plus.owl
http://purl.obolibrary.org/obo/ncbitaxon/subsets/taxslim-disjoint-over-in-taxon.owl

In addition to go-gaf for the load:
http://skyhook.berkeleybop.org/master/ontology/extensions/go-modules-annotations.owl
http://skyhook.berkeleybop.org/master/ontology/extensions/go-taxon-subsets.owl
http://purl.obolibrary.org/obo/eco/eco-basic.owl
http://purl.obolibrary.org/obo/pato.owl
http://purl.obolibrary.org/obo/po.owl
http://purl.obolibrary.org/obo/chebi.owl
http://purl.obolibrary.org/obo/uberon/basic.owl
http://purl.obolibrary.org/obo/wbbt.owl

**Comment by kltm** [comment]
@ukemi @vanaukenk Would it be possible to get your eyes on this to see what is actually needed in the AmiGO GOlr load? I suspect the extensions are driving a lot of this.

**Comment by ukemi** [comment]
Is it safe to assume that we need everything that we need in the editor's version of the ontology plus the annotation information? If so, lots of ontologies are missing like CL, fao, so, and the bridging axioms. @balhoff, do we need all of them?

**Comment by ukemi** [comment]
Probably don't need the bridging axioms etc, but things like the additional anatomies I 'think' would be required.

**Comment by kltm** [comment]
Okay, from the software call today, after the next release, I'll make a go-gaf that is the union and open a PR for somebody with more knowledge. Once in, we can start testing in the pipeline to see if the build works.

**Comment by balhoff** [comment]
Currently having a little issue because the full published version of CHEBI has an unsatisfiable class:

```bash
2020-04-16 21:19:22,430 ERROR org.obolibrary.robot.ReasonerHelper - There are 1 unsatisfiable classes in the ontology.
2020-04-16 21:19:22,430 ERROR org.obolibrary.robot.ReasonerHelper -     unsatisfiable: http://purl.obolibrary.org/obo/CHEBI_143131
```

**Comment by ukemi** [comment]
I believe this is because it can't be both a role and a chemical. 
I have opened a ticket to ask:
https://github.com/ebi-chebi/ChEBI/issues/3770

**Comment by pgaudet** [comment]
what's the status of this?

**Comment by balhoff** [comment]
AmiGO is still pulling in ontologies separately. This is giving us problems like https://github.com/geneontology/go-ontology/issues/23468. @kltm are there some considerations for AmiGO that we need to think about besides just providing one merged ontology?

**Comment by kltm** [comment]
@balhoff If I'm following what you're asking here, I think the answer is: not really.
Ideally, as we approach https://github.com/geneontology/pipeline/issues/27, we would not be taking multiple bites at the ontology apples and would be running self-contained once the ontology had built within the pipeline. Assuming that we had the the same terms (and hopefully some metadata file to control it), I think we'd be good.

**Comment by kltm** [comment]
@pgaudet We're thinking about where this could/should go so that we don't lose it again. This has recently cropped up again wrt https://github.com/geneontology/go-ontology/issues/25177. We're going to have some issues and overhead with this until we resolved it, but it seems to keep slipping through our fingers. Is there an ontology overhead project, or should we slate a special mini-project for this?

**Comment by kltm** [comment]
Can be closed once appearing in `snapshot`s.

**Comment by balhoff** [comment]
I see this file in the latest snapshot.

**Comment by kltm** [comment]
@balhoff Confirming that http://snapshot.geneontology.org/ontology/extensions/go-gaf.owl is the "final" materialized unionized ontology file that we want to be using everywhere. I'll change the GOlr loading over to this in `master` and `snapshot`.

**Comment by balhoff** [comment]
@kltm `go-amigo.owl` is the one.

**Comment by kltm** [comment]
@balhoff Corrected--cheers!

**Comment by kltm** [comment]
Next steps:
- once successful, will put this out to a test server
- will ask for feedback
- if okay, push to `release`, become new final display standard

**Comment by kltm** [comment]
Okay, we're going to switch the `go-ontology-dev` build to look at this "materialized" ontology instead of using the broader set.
